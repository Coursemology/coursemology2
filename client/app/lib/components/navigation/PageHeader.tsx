import { ReactNode } from 'react';
import { useNavigate } from 'react-router-dom';
import ArrowBack from '@mui/icons-material/ArrowBack';
import { AppBar, IconButton, Toolbar, Typography } from '@mui/material';

interface PageHeaderProps {
  title: NonNullable<ReactNode>;
  returnLink?: string;
  toolbars?: ReactNode;
  children?: ReactNode;
}

/**
 * @deprecated Pages should use `Page` and the `title` prop instead.
 */
const PageHeader = (props: PageHeaderProps): JSX.Element => {
  const { title, returnLink, toolbars, children } = props;
  const navigate = useNavigate();

  return (
    <AppBar
      className="border-only-b-neutral-200"
      color="transparent"
      elevation={0}
      position="static"
    >
      <Toolbar className="flex items-center justify-between px-8 py-4">
        <div className="flex min-h-[4rem] w-full items-center">
          {returnLink && (
            <IconButton
              data-testid="ArrowBackIconButton"
              edge="start"
              onClick={(): void => navigate(returnLink)}
            >
              <ArrowBack data-testid="ArrowBack" />
            </IconButton>
          )}

          <Typography className="line-clamp-2" color="inherit" variant="h5">
            {title}
          </Typography>
        </div>

        <div className="flex shrink-0 items-center">
          {children ?? toolbars ?? null}
        </div>
      </Toolbar>
    </AppBar>
  );
};

export default PageHeader;
