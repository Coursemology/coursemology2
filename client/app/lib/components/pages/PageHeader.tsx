import { FC, ReactElement } from 'react';
import { useNavigate } from 'react-router-dom';
import { IconButton } from '@mui/material';
import ArrowBack from '@mui/icons-material/ArrowBack';
import TitleBar from 'lib/components/TitleBar';

interface Props {
  title: string;
  returnLink?: string;
  toolbars?: ReactElement[];
}

const PageHeader: FC<Props> = (props) => {
  const { title, returnLink, toolbars } = props;
  const navigate = useNavigate();

  return (
    <TitleBar
      title={title}
      iconElementRight={toolbars && toolbars.length > 0 ? toolbars : null}
      iconElementLeft={
        returnLink ? (
          <IconButton onClick={() => navigate(returnLink)}>
            <ArrowBack htmlColor="white" />
          </IconButton>
        ) : null
      }
    />
  );
};

export default PageHeader;
