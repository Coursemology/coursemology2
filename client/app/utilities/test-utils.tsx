// eslint-disable-next-line import/no-extraneous-dependencies
import { render, RenderResult } from '@testing-library/react';

import TestApp, { CustomRenderOptions } from './TestApp';

const customRender = (
  ui: JSX.Element,
  options?: CustomRenderOptions,
): RenderResult => render(<TestApp {...options}>{ui}</TestApp>);

// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';
export { customRender as render };
