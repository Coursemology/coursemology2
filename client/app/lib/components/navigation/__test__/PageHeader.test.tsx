import { render, RenderResult } from 'test-utils';

import PageHeader from '../PageHeader';

let documentBody: RenderResult;

describe('<PageHeader />', () => {
  describe('when there is no return link', () => {
    beforeEach(() => {
      documentBody = render(<PageHeader title="Test Title 1" />);
    });

    it('renders TitleBar and shows the title', () => {
      expect(documentBody.getByText('Test Title 1')).toBeVisible();
    });

    it('does not show the return/back button', () => {
      expect(
        documentBody.queryByTestId('ArrowBackIconButton'),
      ).not.toBeInTheDocument();
    });

    it('matches the snapshot', () => {
      const { baseElement } = documentBody;
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('when there is a return link', () => {
    beforeEach(() => {
      documentBody = render(
        <PageHeader returnLink="some/link" title="Test Title 2" />,
      );
    });

    it('renders TitleBar and shows the title', () => {
      expect(documentBody.getByText('Test Title 2')).toBeVisible();
    });

    it('shows the return/back button', () => {
      expect(documentBody.getByTestId('ArrowBackIconButton')).toBeVisible();
    });

    it('matches the snapshot', () => {
      const { baseElement } = documentBody;
      expect(baseElement).toMatchSnapshot();
    });
  });
});
