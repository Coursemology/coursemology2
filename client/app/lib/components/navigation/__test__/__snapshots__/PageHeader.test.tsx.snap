// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> when there is a return link matches the snapshot 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic border-only-b-neutral-200 css-1na8te8-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular flex items-center justify-between px-8 py-4 css-129s0w6-MuiToolbar-root"
      >
        <div
          class="flex min-h-[4rem] w-full items-center"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-14cf8m5-MuiButtonBase-root-MuiIconButton-root"
            data-testid="ArrowBackIconButton"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hgpioi-MuiSvgIcon-root"
              data-testid="ArrowBack"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h5
            class="MuiTypography-root MuiTypography-h5 line-clamp-2 css-1h0fw4b-MuiTypography-root"
          >
            Test Title 2
          </h5>
        </div>
        <div
          class="flex shrink-0 items-center"
        />
      </div>
    </header>
    <div
      class="Toastify"
    />
  </div>
</body>
`;

exports[`<PageHeader /> when there is no return link matches the snapshot 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic border-only-b-neutral-200 css-1na8te8-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular flex items-center justify-between px-8 py-4 css-129s0w6-MuiToolbar-root"
      >
        <div
          class="flex min-h-[4rem] w-full items-center"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 line-clamp-2 css-1h0fw4b-MuiTypography-root"
          >
            Test Title 1
          </h5>
        </div>
        <div
          class="flex shrink-0 items-center"
        />
      </div>
    </header>
    <div
      class="Toastify"
    />
  </div>
</body>
`;
